python/tank/util/filesystem.py

```diff
@@ -171,7 +171,15 @@ def copy_file(src, dst, permissions=0o666):
     :param permissions: Permissions to use for target file. Default permissions will
                         be readable and writable for all users.
     """
-    shutil.copy(src, dst)
+    if is_windows():
+        windows_src = open(src, mode="rb")
+        windows_dst = open(dst, mode="wb")
+        shutil.copyfileobj(windows_src, windows_dst, length=16 * 1024 * 1024)
```

Comment:
If something goes wrong on this line, we'll be leaving the files open until garbage collection kicks in. Might be better to use context managers:
```
with open(src, mode="rb") as windows_src:
    with open(dst, mode="wb") as windows_dst:
        shutli.copyfileobj(...)
```

Reply:
Hey @eshokrgozar ,

- I added a check if `dst` is a directory and change it to a filename.
- For the mode bits, it is weird on Windows; the next line is a call to `os.chmod` so that seems like it gets thrown-away, am I missing something there?
- I tested this override with symlinks on Windows (that do exist) and it follows them to the file by default.
- Switched to context managers, thanks for that!

Let me know if there is anything else that comes to mind with this one, cheers!

Reply:
Also adding @staceyoue to the cc of this one. We should test this on a near-full disk that would trigger a `disk full` error to ensure the exception handling works properly.